app/controllers/api/usage_limits_controller.rb

```diff
@@ -55,20 +57,10 @@ def destroy
   private
 
   def find_plan
-    @plan = find_application_plan || find_end_user_plan
-  end
-
-  def find_application_plan
     id = params[:application_plan_id]
-    current_account.application_plans.find(id) if id
+    @plan = current_account.application_plans.find(id) if id
   end
```

Comment:
Personally I would do this:
```ruby
@plan = current_account.application_plans.find(params[:application_plan_id])
```

Because if that param is not sent, right now it will raise an error with NoMethodError for NilClass and will return an ugly 5XX to the user. But with what I am suggesting, it returns a 404, which is what it should do in this case considering that everything inside this controller now is inside this plan.